base/MAPL_Generic.F90

```diff
@@ -5371,6 +5371,20 @@ subroutine MAPL_ESMFStateWriteToFile(STATE,CLOCK,FILENAME,FILETYPE,MPL,HDR, oCli
     !real(kind=ESMF_KIND_R8),save ::  total_time = 0.0d0
     !logical                               :: amIRoot
     !type (ESMF_VM)                        :: vm
+    logical :: empty
+    class(Logger), pointer :: lgr
+
+! Check if state is empty. If "yes", simply return
+    empty = MAPL_IsStateEmpty(state, __RC__)
+    if (empty) then
+       if (MAPL_Am_I_Root()) then
+          call MAPL_GetLogger(mpl, lgr, __RC__)
+          call lgr%warning('Checkpoint '//trim(filename) //&
+               ' requested, but state is empty. Ignored...')
+       end if
+
+       _RETURN(ESMF_SUCCESS)
+    end if
```

Comment:
You should introduce a procedure `handle_empty()` that takes one argument `empty`.    This eliminates the duplication. Were it not for the multiple error possibilities, I'd say:
```f90
if (MAPL_isStateEmpty(state)) call handle_empty()
```
instead you'll need to split it into two statements:
```f90
empty = ...
if (empty) then
    call handle_empty(__RC__)
    _RETURN(ESMF_SUCCESS)
end if
```
A case where lack of exceptions makes this much more verbose than it should be.  Sih.

Reply:
A procedure cannot save us - the point is to return from the caller (and log a warning)

Reply:
We simply do not want to execute the rest of this subroutine. I cannot see how you do it by calling a procedure. Only way to save typing would be a macro, but I saw little value in that

Reply:
or maybe
```f90
empty = MAPL_isStateEmpty(state, __RC__)
if (empty) __RETURN(status)
```